src/java/org/apache/cassandra/utils/memory/LittleEndianMemoryUtil.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils.memory;
+
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.cassandra.utils.Architecture;
+
+public class LittleEndianMemoryUtil extends MemoryUtil
+{
+    public static int getUnsignedShort(long address)
+    {
+        if (Architecture.IS_UNALIGNED || (address & 0b1) == 0L)
+            return (Architecture.BIG_ENDIAN ? Short.reverseBytes(unsafe.getShort(address)) : unsafe.getShort(address)) & 0xffff;
+        else
+            return getShortByByte(address) & 0xffff;
+    }
+
+    public static int getInt(long address)
+    {
+        if (Architecture.IS_UNALIGNED || (address & 0b11) == 0L)
+            return Architecture.BIG_ENDIAN ? Integer.reverseBytes(unsafe.getInt(address)) : unsafe.getInt(address);
+        else
+            return getIntByByte(address);
+    }
+
+    public static long getLong(long address)
+    {
+        if (Architecture.IS_UNALIGNED || (address & 0b111) == 0L)
+            return Architecture.BIG_ENDIAN ? Long.reverseBytes(unsafe.getLong(address)) : unsafe.getLong(address);
+        else
+            return getLongByByte(address);
+    }
+
+    public static void setShort(long address, short s)
+    {
+        if (Architecture.IS_UNALIGNED)
+            unsafe.putShort(address, Architecture.BIG_ENDIAN ? Short.reverseBytes(s) : s);
+        else
+            putShortByByte(address, s);
+    }
+
+    public static void setInt(long address, int l)
+    {
+        if (Architecture.IS_UNALIGNED)
+            unsafe.putInt(address, Architecture.BIG_ENDIAN ? Integer.reverseBytes(l) : l);
+        else
+            putIntByByte(address, l);
+    }
+
+    public static void setLong(long address, long l)
+    {
+        if (Architecture.IS_UNALIGNED)
```

Comment:
Shouldn't we have the aligned address option (e.g. `|| (address & 0b111) == 0L`) to use a direct store in the writes as well?

Reply:
thank you for noticing the gap, done